sw-src.js

```diff
@@ -55,99 +45,57 @@ self.registration.addEventListener("updatefound", function (e) {
     });
   });
 });
-// async function cacheAudiosFiles(file, cacheName, length) {
-//   ///awt
-//   await caches.open(cacheName).then(function (cache) {
-//     cache
-//       .add(
-        // self.location.href.includes("https://feedthemonsterdev.curiouscontent.org")
-        //   ? file.slice(
-        //       0,
-        //       file.indexOf("/feedthemonster") + "/feedthemonster".length
-        //     ) +
-        //       "dev" +
-        //       file.slice(
-        //         file.indexOf("/feedthemonster") + "/feedthemonster".length
-        //       )
-        //   : file
-//       )
-//       .finally(() => {
-//         number = number + 1;
-//         self.clients.matchAll().then((clients) => {
-//           clients.forEach((client) => {
-//             if ((number / (length * 5)) * 100 < 101) {
-//               client.postMessage({
-//                 msg: "Loading",
-//                 data: Math.round((number / (length * 5)) * 100),
-//               });
-//             } else {
-//               client.postMessage({
-//                 msg: "Loading",
-//                 data: Math.round(100),
-//               });
-//             }
-//           });
-//         });
-//       });
-//   });
-// }
+
 async function cacheLangAssets(file, cacheName) {
-  await caches.open(cacheName).then((cache) => {
-    cache.add(file);
-  });
+  const cache = await caches.open(cacheName);
+  const cachedResponse = await cache.match(file);
+
+  if (!cachedResponse) {
+    await cache.add(file);
+    console.log('Cached File:', file);
+  } else {
+    console.log('File already cached, skipping:', file);
+  }
 }
 async function getCacheName(language) {
-  ///awt
- await caches.keys().then((cacheNames) => {
+  await caches.keys().then((cacheNames) => {
     cacheNames.forEach(async (cacheName) => {
       await getALLAudioUrls(cacheName, language);
     });
   });
 }
 
 async function getALLAudioUrls(cacheName, language) {
-  // await cacheCommonAssets(language);
-  let audioList = [];
-  audioList.push("/lang/" + language + "/ftm_" + language + ".json");
+  let audioList = new Set(); // Use Set to filter duplicates
+  audioList.add("/lang/" + language + "/ftm_" + language + ".json");
+
   fetch("./lang/" + language + "/ftm_" + language + ".json", {
```

Comment:
this could improve into `./lang/${language}/ftm_${language}.json`

Reply:
sure will consider this